raiden/network/transport/matrix.py

```diff
@@ -1349,52 +1387,55 @@ def _get_room_ids_for_address(
         If filter_private=None, filter according to self._private_rooms
         """
         address_hex: AddressHex = to_checksum_address(address)
-        room_ids = self._client.account_data.get(
-            'network.raiden.rooms',
-            {},
-        ).get(address_hex)
-        if not room_ids:  # None or empty
-            room_ids = list()
-        if not isinstance(room_ids, list):  # old version, single room
-            room_ids = [room_ids]
-
-        if filter_private is None:
-            filter_private = self._private_rooms
-        if not filter_private:
-            # existing rooms
-            room_ids = [
-                room_id
-                for room_id in room_ids
-                if room_id in self._client.rooms
-            ]
-        else:
-            # existing and private rooms
-            room_ids = [
-                room_id
-                for room_id in room_ids
-                if room_id in self._client.rooms and self._client.rooms[room_id].invite_only
-            ]
+        with self._account_data_lock:
+            room_ids = self._client.account_data.get(
+                'network.raiden.rooms',
+                {},
+            ).get(address_hex)
+            self.log.debug('acct data get', room_ids=room_ids, for_address=address_hex)
+            if not room_ids:  # None or empty
+                room_ids = list()
+            if not isinstance(room_ids, list):  # old version, single room
+                room_ids = [room_ids]
+
+            if filter_private is None:
+                filter_private = self._private_rooms
+            if not filter_private:
+                # existing rooms
+                room_ids = [
+                    room_id
+                    for room_id in room_ids
+                    if room_id in self._client.rooms
+                ]
+            else:
+                # existing and private rooms
+                room_ids = [
+                    room_id
+                    for room_id in room_ids
+                    if room_id in self._client.rooms and self._client.rooms[room_id].invite_only
+                ]
+
+            return room_ids
+
+    def _leave_unused_rooms(self, _address_to_room_ids: Dict[AddressHex, List[_RoomID]]):
+        """
+        Checks for rooms we've joined and which partner isn't health-checked and leave.
+
+        **MUST** be called from a context that holds the `_account_data_lock`.
+        """
+        assert self._account_data_lock.locked(), \
```

Comment:
use our convention:

```python
msg = '_leave_unused_rooms called without account data lock'
assert self._account_data_lock.locked(), msg
```

Reply:
Ah, right, thanks. That seemed wrong somehow.